pages/index.tsx

```diff
@@ -30,66 +31,58 @@ const Home: NextPage = () => {
   }
 
   function countNodes(nodes: NodeX[]): number {
-    if (nodes === null || nodes === undefined || nodes.length === 0) return 0;
-
-    let count = nodes.length;
-
-    nodes.forEach((child: NodeX) => {
-      count += countNodes(child.children);
-    });
-
-    return count;
+    if (!nodes) return 0;
+    return nodes.length;
   }
 
   function toggleCheck(node: NodeX): void {
     if (!node) return;
-
-    node.checked = !node.checked;
-    forceRerender();
+    setNodes(
+      nodes.map((n) => (n === node ? { ...n, checked: !n.checked } : n))
+    );
```

Comment:
Its a good practice to use the previous state that you get from the set function in the state. Doing this makes it so that there is no risk of `nodes`, in this case, not being stale when its being updating. Doing that would look like this:
```suggestion
    setNodes((prevNodes) =>
      prevNodes.map((n) => (n === node ? { ...n, checked: !n.checked } : n)),
    );
```